### ISSUE TITLE
Sequel Pro doesn't recognise the foreign characters and emoji flags

### ISSUE THREAD

**Issue by gusbemacbe** [sequelpro/sequelpro#2536]
Hello, 

I saved a SQL file from MySQL Workbench and TextMate and open it with Sequel Pro. I found many foreign characters not recognised by this app. It is written in Portuguese and Georgian. Check the screenshot:

![Sequel Pro with Not Recognised Foreign Characters](https://cloud.githubusercontent.com/assets/2257024/16972392/58fa97f6-4e01-11e6-9efc-7de174b2b64e.png)

Thank you for your attention, help, patience and understanding!

Gustavo

**Comment by gusbemacbr** [comment]
Hello, 

I saved a SQL file from MySQL Workbench and TextMate and open it with Sequel Pro. I found many foreign characters not recognised by this app. It is written in Portuguese and Georgian. Check the screenshot:

![Sequel Pro with Not Recognised Foreign Characters](https://cloud.githubusercontent.com/assets/2257024/16972264/69a1bc52-4e00-11e6-9ee7-1229986dd28c.png) 

Thank you for your attention, help, patience and understanding!

Gustavo

**Comment by gusbemacbr** [comment]
Hello, 

I saved a SQL file from MySQL Workbench and TextMate and open it with Sequel Pro. I found many foreign characters not recognised by this app. It is written in Portuguese and Georgian. Check the screenshot:

![Sequel Pro with Not Recognised Foreign Characters](https://cloud.githubusercontent.com/assets/2257024/16972264/69a1bc52-4e00-11e6-9ee7-1229986dd28c.png) 

Thank you for your attention, help, patience and understanding!

Gustavo

**Comment by dmoagx** [comment]
That screenshot is not very useful.
You either
* used a charset in the file that was wrongly guessed by the autodetection
* have a byte sequence in the file that is not valid for the desired charset (in which case Sequel Pro falls back to latin1)

Use Console.app to search for any messages from Sequel Pro.

**Comment by gusbemacbr** [comment]
I do not know is it the right log:

[Sequel Pro - System Report.zip](https://github.com/sequelpro/sequelpro/files/374951/Sequel.Pro.-.System.Report.zip)

**Comment by dmoagx** [comment]
No, that is not helpful for this issue.
If possible, send the file you have problems with. You can also send it via mail to us if you don't want to share it here: http://www.sequelpro.com/docs/Contact_the_developers

**Comment by gusbemacbr** [comment]
But this was the only log that I found in the Console app. I do not know which exact log I should send to you or to the developers via e-mail.

**Comment by dmoagx** [comment]
Select "All messages" in the left pane of Console.app and then use the search field in the upper right corner to search for "Sequel Pro".

**Comment by slackero** [comment]
If you do not know the encoding of your SQL dump take [TextWrangler](http://www.barebones.com/products/textwrangler/). You find the charset information in the status bar, in my case Unicode (UTF-8). That is the easiest way I know.
![textwrangler-charset-statusbar](https://cloud.githubusercontent.com/assets/15139/17579506/bdffb428-5f96-11e6-8966-e47bfdb990cb.png)

**Comment by gusbemacbr** [comment]
@dmoagx 

Sorry for my absence. I was busy with the college.

I did what you suggested. But there are many messages. Which message should I select to save file for you?

![Sequel Pro at the Console.app](https://cloud.githubusercontent.com/assets/2257024/17641114/868c4af2-60e9-11e6-8d7b-990350b58d53.png)

**Comment by gusbemacbr** [comment]
@slackero 

It is still the same like yours:

![SQL at the TextWrangler app](https://cloud.githubusercontent.com/assets/2257024/17641156/b37b06f6-60ea-11e6-9266-db6c33c4bb24.png)

Ah, I have just changed from SQL Generic to SQL (MySQL) and UTF-8 General to UTF-8 with BOM, it worked in Sequel Pro:

![The foreign characters appeared working in Sequel Pro](https://cloud.githubusercontent.com/assets/2257024/17641177/6669d8d2-60eb-11e6-91d3-696d6d0600c1.png)


Not just TextWrangler, it worked also at TextMate (it reads the emojis) and MySQL Workbench (both work with SQL Generic and UTF-8 without BOM:
![On TextMate](https://cloud.githubusercontent.com/assets/2257024/17641182/92c9989a-60eb-11e6-8cae-f8c43bfa52fc.png)

It seems that Sequel Pro doesn't support SQL Generic and UTF-8 without BOM.

**Comment by slackero** [comment]
@gusbemacbr Your file wasn't well formed, by opening and trying you have found a way around your problem.

**Comment by dmoagx** [comment]
> It seems that Sequel Pro doesn't support SQL Generic and UTF-8 without BOM.

"SQL Generic" is just the language setting TextWrangler uses for syntax highlighting. That setting has zero effect on the file contents.

Also Sequel Pro does support UTF8 with and without BOM, but without BOM Sequel Pro has to guess the encoding and it seems like it wrongly guessed it as cp855 or similar.

**How** did you open the file in Sequel Pro?

**Comment by gusbemacbr** [comment]
@dmoagx 

> How did you open the file in Sequel Pro?

1. I go directly to the `.../Documents/SQL/biblioteca_database.sql` in the Finder. 
2. I click the `biblioteca_database.sql` and open it with the mouse right button to select Sequel Pro.
3. Opening Sequel Pro, I type connection details and connect.

Before, “autodetect” was always default encoding in the Sequel Pro.

As the foreign characters weren't recognised, I had to change the default encoding from “autodetect” to “UTF-8 Unicode BMP”, closed Sequel Pro and open again that file, but the foreign characters still weren't recognised either. 

As @slackero suggested, I set the database language and encoding at the TextWrangler and saved the file. I open the file with Sequel Pro, from Finder and the foreign characters appeared normally, with both “autodetect” and “UTF-8 Unicode BMP”. 

> Also Sequel Pro does support UTF8 with and without BOM, but without BOM Sequel Pro has to guess the encoding and it seems like it wrongly guessed it as cp855 or similar.

I believe it is the file bug. 

As I have database classes at my college, I always save the files from MySQL Workbench on a college computer to my USB drive and move them to my Mac to open with Sequel Pro.

**Comment by slackero** [comment]
Use File > Open in Sequel Pro. Then you can choose the encoding by clicking the button [Options]

![bildschirmfoto 2016-08-13 um 20 06 33](https://cloud.githubusercontent.com/assets/15139/17644802/779d8b98-6191-11e6-8fbd-413fdfe5982e.png)

**Comment by gusbemacbr** [comment]
Ah, @slackero, I made a test, I changed again to SQL Generic and UTF-8 without BOM in TextWrangler and I did what you suggested. I selected the file to select UTF-8. The foreign characters appeared normally. 

![Selecting SQL and set the option of encoding](https://cloud.githubusercontent.com/assets/2257024/17645389/29a577ea-617b-11e6-8e36-230c8517a43d.png)

![The foreign characters worked now in Sequel Pro, with SQL Generic and UTF-8 without BOM](https://cloud.githubusercontent.com/assets/2257024/17645391/4551b4a4-617b-11e6-80a5-da51c3205fbd.png)

Interesting, if I **did not select** the file **to enable** the option of encoding, the foreign characters would not be working.
